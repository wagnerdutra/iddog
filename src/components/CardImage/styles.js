import styled, { css } from 'styled-components';
import gifLoading from 'Root/assets/loader.gif';

const CardStyle = styled.div`
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  width: ${props => props.width}px;
  height: ${props => props.height}px;
  margin: ${props => props.margin}px;
  border-radius: 2%;

  img {
    transition: all 0.3s;
    width: 100%;
    height: 100%;
    background-position: 50% 50%;
    background-repeat: no-repeat;
    border-radius: 2%;
    background: transparent url(${gifLoading}) center no-repeat;
    transition: all 0.2s ease-in-out;

    ${props =>
      props.hover &&
      css`
        &:hover {
          box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2);
          transform: scale(1.05);
        }
      `}
  }

  ${props =>
    props.cursor &&
    css`
      cursor: pointer;
    `}
`;

export { CardStyle };
