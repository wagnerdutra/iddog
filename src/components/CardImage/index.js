import React, { useRef } from 'react';
import propTypes from 'prop-types';

import { CardStyle } from './styles';

function Card({ src, width, height, cursor, margin, onClick, hover }) {
  const containerRef = useRef();

  function handleContainerClick(e) {
    if (e.target !== containerRef.current) {
      onClick(e.target);
      return;
    }
    onClick(e.target.firstChild);
  }

  return (
    <CardStyle
      width={width}
      height={height}
      cursor={cursor ? 1 : 0}
      margin={margin}
      hover={hover}
      onClick={handleContainerClick}
      ref={containerRef}
    >
      <img src={src} alt={src} />
    </CardStyle>
  );
}

Card.propTypes = {
  src: propTypes.string.isRequired,
  width: propTypes.number,
  height: propTypes.number,
  margin: propTypes.number,
  cursor: propTypes.bool,
  onClick: propTypes.func,
  hover: propTypes.bool
};

Card.defaultProps = {
  width: 100,
  height: 100,
  cursor: false,
  margin: 10,
  onClick: () => {},
  hover: true
};

export default Card;
