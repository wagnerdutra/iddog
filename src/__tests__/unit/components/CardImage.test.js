import React from 'react';

import { render, fireEvent } from 'react-testing-library';

import CardImage from 'Components/CardImage';

describe('<CardImage />', () => {
  it('should render correctly', () => {
    const { container } = render(<CardImage src="teste" />);
    expect(container).toMatchSnapshot();
  });

  it('should handle image click', () => {
    const onClick = jest.fn();
    const { container } = render(<CardImage src="teste" onClick={onClick} />);
    fireEvent.click(container.firstChild);
    fireEvent.click(container.querySelector('img'));
    expect(onClick).toHaveBeenCalledTimes(2);
  });
});
