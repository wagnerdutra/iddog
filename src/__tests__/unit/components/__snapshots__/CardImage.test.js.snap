// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardImage /> should render correctly 1`] = `
.c0 {
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
  width: 100px;
  height: 100px;
  margin: 10px;
  border-radius: 2%;
  position: relative;
  overflow: hidden;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c0:hover {
  box-shadow: 0 8px 16px 0 rgba(0,0,0,0.2);
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
}

.c0 img {
  position: absolute;
  height: 100%;
  width: auto;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  border-radius: 2%;
  background-position: 50% 50%;
  background-repeat: no-repeat;
  background: transparent url(test-file-stub) center no-repeat;
}

<div>
  <div
    class="c0"
    cursor="0"
    height="100"
    width="100"
  >
    <img
      alt="teste"
      src="teste"
    />
  </div>
</div>
`;
